src/Executables/FindHorizons/FindHorizons.hpp

```diff
@@ -0,0 +1,325 @@
+        spatial_christoffel_second_kind, mesh, inv_jacobian);
+    auto& spatial_ricci =
+        get<gr::Tags::SpatialRicci<Dim, Frame::Inertial>>(*target_vars);
+    gr::ricci_tensor(make_not_null(&spatial_ricci),
+                     spatial_christoffel_second_kind,
+                     deriv_spatial_christoffel_second_kind);
+  }
+};
+
+template <size_t Dim, ah::ObjectLabel Label>
+struct ApparentHorizon
+    : tt::ConformsTo<intrp::protocols::InterpolationTargetTag> {
+  static std::string name() { return "Ah" + ah::name(Label); }
+  using temporal_id = ::Tags::Time;
+  using compute_target_points =
+      intrp::TargetPoints::ApparentHorizon<ApparentHorizon, Frame::Inertial>;
+  using post_interpolation_callback =
+      intrp::callbacks::FindApparentHorizon<ApparentHorizon, Frame::Inertial>;
+  using horizon_find_failure_callback =
+      intrp::callbacks::ErrorOnFailedApparentHorizon;
+  using post_horizon_find_callbacks = tmpl::list<
+      intrp::callbacks::ObserveTimeSeriesOnSurface<
+          ::ah::tags_for_observing<Frame::Inertial>, ApparentHorizon>,
+      intrp::callbacks::ObserveSurfaceData<::ah::surface_tags_for_observing,
+                                           ApparentHorizon, ::Frame::Inertial>>;
+
+  using vars_to_interpolate_to_target =
+      ::ah::vars_to_interpolate_to_target<Dim, Frame::Inertial>;
+  using compute_vars_to_interpolate = ComputeHorizonVolumeQuantities<Dim>;
+  using compute_items_on_target =
+      ::ah::compute_items_on_target<Dim, Frame::Inertial>;
+};
+
+template <size_t Dim, bool TwoHorizons>
+struct Metavariables {
+  static constexpr size_t volume_dim = Dim;
+
+  static constexpr Options::String help{
+      "Find apparent horizons in volume data."};
+
+  // A placeholder system for the domain creators
+  struct system {};
+
+  using AhA = ApparentHorizon<Dim, ah::ObjectLabel::A>;
+  using AhB = ApparentHorizon<Dim, ah::ObjectLabel::B>;
+  static constexpr bool two_horizons = TwoHorizons;
+
+  struct domain : tt::ConformsTo<::domain::protocols::Metavariables> {
+    static constexpr bool enable_time_dependent_maps = false;
+  };
+
+  using const_global_cache_tags = tmpl::list<>;
+
+  using adm_vars = tmpl::list<
+      gr::Tags::Lapse<DataVector>,
+      gr::Tags::Shift<Dim, Frame::Inertial, DataVector>,
+      gr::Tags::SpatialMetric<Dim, Frame::Inertial, DataVector>,
+      gr::Tags::ExtrinsicCurvature<Dim, Frame::Inertial, DataVector>>;
+
+  using interpolator_source_vars =
+      typename ComputeHorizonVolumeQuantities<Dim>::required_src_tags;
+  using interpolation_target_tags = tmpl::flatten<
+      tmpl::list<AhA, tmpl::conditional_t<two_horizons, AhB, tmpl::list<>>>>;
+  using temporal_id = ::Tags::Time;
+
+  struct factory_creation
+      : tt::ConformsTo<Options::protocols::FactoryCreation> {
+    using factory_classes =
+        tmpl::map<tmpl::pair<DomainCreator<Dim>, domain_creators<Dim>>>;
+  };
+
+  using element_array = elliptic::DgElementArray<
```

Comment:
Not a change request: would switching this to evolution allow us to more easily read different frames and variables? I'm not sure how tied the elliptic one is to grid==inertial. The evolution code can operate with or without a time dependent map.

Reply:
Yep, I switched this to evolution on https://github.com/nilsvu/spectre/tree/find_horizons_exec_gridframe, and the Domain init action too. It's a drop-in replacement, no problem. The elliptic one is a bit leaner (fewer compute items etc), so I chose that one for now. For moving meshes this can be switched, but currently the interpolation framework doesn't support both static and time-dependent domains at runtime.

Reply:
Okay, awesome! That makes sense, thank you :)